### PR TITLE
consolidate docs directory

### PR DESCRIPTION
**Pull request by bryevdv**
@tcmetzger I was going to open a discussion but I figured it would be easier to just discuss a concrete PR. 

This PR moves the `sphinx` directory to `docs/bokeh`. I think this would be good for a few reasons:

* Similar to `src` move, if we split out more packages, e.g. a Bokeh compiler package under `src`, then we might probably want to distinguish the package docs builds separately as well, e.g. `docs/bokeh-compiler` or whatever
* Eventual dedicated BokehJS docs could live under docs dir
* "docs" is a much less jingo-y name that will be more immediately meaningful to more people than than "sphinx"
* GH docs state that some of the top-level files (CoC, security, etc) can live under a `docs` dir, we can clean up the top-level even more

If you agree, then there is probably a bit more light updates in the dev guide (plus any issues that show up in CI) let me know what you think. 

One of the other things I appreciated in working this up is that the path handling for sphing extensions, gallery, example handling, etc. is completely scattered and non-unified. I would clean all that up in a separate PR though.

### CODE REVIEW

**Comment by tcmetzger** [comment]
This makes a lot of  sense, but might be a bit of a can of worms. Let's see what happens! 
I'd only ask that we merge https://github.com/bokeh/bokeh/pull/12242 first, as this PR updates quite a few files in the old docs structure.

**Comment by bryevdv** [comment]
@tcmetzger It's possible that https://github.com/bokeh/bokeh/pull/12242 will rebase cleanly on this. In any case I'm happy to order the PRs whatever is most convenient for you. Note that @mattpap merged the src layout PR so there are currently conflicts in #12242 due to that. @mattpap did you just have to do a simple `git rebase branch-3.0` to update your layout PR? If not please advise the specific command @tcmetzger can use on his PR.

**Comment by mattpap** [comment]
> did you just have to do a simple `git rebase branch-3.0` to update your layout PR? 

For the most part. Some manual intervention was needed for removed or renamed files.

**Comment by bryevdv** [comment]
@tcmetzger what are your thoughts on merging this?

**Comment by tcmetzger** [comment]
We'll still need to adjust the info in documentation.rst (line 187, e.g.)!

**Comment by tcmetzger** [comment]
I have updated some paths with my most recent commit.

@bryevdv Please take a look at some of the scripts where we might need to update more paths. For example:

* deploy.py:51
* milestone.py:313, milestone.py:316
* bokeh_plot.py:278

File: README.md [hunk 1]
```diff
@@ -291,7 +291,7 @@ Community support is available on the [Project Discourse](https://discourse.boke
 
 If you would like to contribute to Bokeh, please review the [Contributor Guide](https://docs.bokeh.org/en/latest/docs/dev_guide.html) and [request an invitation to the Bokeh Dev Slack workspace](https://slack-invite.bokeh.org/).
 
-*Note: Everyone who engages in the Bokeh project's discussion forums, codebases, and issue trackers is expected to follow the [Code of Conduct](https://github.com/bokeh/bokeh/blob/main/CODE_OF_CONDUCT.md).*
+*Note: Everyone who engages in the Bokeh project's discussion forums, codebases, and issue trackers is expected to follow the [Code of Conduct](https://github.com/bokeh/bokeh/blob/main/docs/CODE_OF_CONDUCT.md).*
```
Comment (bryevdv) [inline README.md:294]: we will probably need to make `main` point at the 3.0 release tag for this to work, but that's also probably a reasonable thing we should do

File: docs/bokeh/source/_templates/sections/footer.html [hunk 2]
```diff
@@ -77,7 +77,7 @@ <h6>Contribute</h6>
           project needs.
         </p>
         <p class="m-0">
-          {{ copyright }} The website content uses the <a href="https://github.com/bokeh/demo.bokeh.org/blob/master/LICENSE.txt">BSD License</a> and is covered by the <a href="https://github.com/bokeh/bokeh/blob/master/CODE_OF_CONDUCT.md">Bokeh Code of Conduct</a>.
+          {{ copyright }} The website content uses the <a href="https://github.com/bokeh/demo.bokeh.org/blob/master/LICENSE.txt">BSD License</a> and is covered by the <a href="https://github.com/bokeh/bokeh/blob/master/docs/CODE_OF_CONDUCT.md">Bokeh Code of Conduct</a>.
```
Comment (bryevdv) [inline docs/bokeh/source/_templates/sections/footer.html:80]: master -> main

File: docs/bokeh/source/docs/dev_guide.rst [hunk 3]
```diff
@@ -245,7 +245,7 @@ updates and news about Bokeh. And we always appreciate it if you tag Bokeh's
 accounts when you talk about anything that you made with Bokeh!
 
 
-.. _Code of Conduct: https://github.com/bokeh/bokeh/blob/master/CODE_OF_CONDUCT.md
+.. _Code of Conduct: https://github.com/bokeh/bokeh/blob/master/docs/CODE_OF_CONDUCT.md
```
Comment (bryevdv) [inline docs/bokeh/source/docs/dev_guide.rst:248]: master -> main

**Comment by bryevdv** [comment]
@tcmetzger thanks I left a few comments

**Comment by tcmetzger** [comment]
@bryevdv Thanks! I updated the two mentions of `master` to `main`. 
When I search for `bokeh/blob/master` across the entire repo, I get 24 more results. Should we take this opportunity to update them all?

And can you look into the scripts I mentioned above?

**Comment by bryevdv** [comment]
@tcmetzger thanks I have pushed changes for all those files 

FYI I also noticed that the 2.4.3 changelog entry was never ported to `branch-3.0` Why don't we have a follow-on PR for that, and for more master->main updates (this PR is big enough)

**Comment by bryevdv** [comment]
@tcmetzger any objection to merging this as-is?

**Review by tcmetzger — APPROVED** [review]
LGTM!

**Comment by bryevdv** [comment]
@mattpap do you expect this will conflict or cause issues for your PR?

**Comment by mattpap** [comment]
> do you expect this will conflict or cause issues for your PR?

It should rebase fairly cleanly.

**Pull request merged**